File: models/usermanagement/Organization.js
import mongoose from "mongoose";

const OrganizationSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true
  },
  registrationNumber: String,
  taxId: String,
  email: String,
  phone: String,
  website: String,
  address: {
    street: String,
    city: String,
    state: String,
    zipCode: String,
    country: String
  },
  logo: String,
  settings: {
    timezone: {
      type: String,
      default: "UTC"
    },
    currency: {
      type: String,
      default: "INR"
    },
    workHours: {
      start: String,
      end: String
    }
  },
  hospitalId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Hospital"
  }
}, { timestamps: true });

export default mongoose.model("Organization", OrganizationSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/Organization.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "organizations")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Organization extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Column(name = "registration_number")
    private String registrationNumber;

    @Column(name = "tax_id")
    private String taxId;

    private String email;
    private String phone;
    private String website;

    @Embedded
    private AddressInfo address;

    private String logo;

    @Embedded
    private SettingsInfo settings;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id")
    private Hospital hospital;

    @Embeddable
    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class AddressInfo {
        private String street;
        private String city;
        private String state;
        @Column(name = "zip_code")
        private String zipCode;
        private String country;
    }

    @Embeddable
    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class SettingsInfo {
        private String timezone = "UTC";
        private String currency = "INR";
        @Embedded
        private WorkHoursInfo workHours;
    }

    @Embeddable
    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class WorkHoursInfo {
        private String start;
        private String end;
    }
}
